### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mhwatson**

### CODE REVIEW

File: src/integrations/airtable.ts [hunk 1]
```diff
@@ -16,19 +16,47 @@ const base = new Airtable({apiKey: AIRTABLE_KEY}).base(AIRTABLE_BASE);
...
+            if (record === undefined) {
+                return rej(new Error("Record is undefined"));
             }
 
             return res(record.fields);
         });
     });
 }
+
+export const write = async (table, {
```
Comment (kshahzada) [inline src/integrations/airtable.ts:36]: this shouldn't be written as a general "write" function where you can define the table name, if you are going to have rating, etc defined as parameters (they only apply to one table)

File: src/services/packages/routes.ts [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+  getPouchRedirectToRoaster,
+  postRating,
+} from "./controllers";
 
 export const router = express.Router();
 
-router.get("/v1/pouches/:pouch_id/redirect", getPouchRedirect);
-router.get("/v1/pouches/:pouch_id", getPouch);
+// Just handle all versions downstream
+router.get("/:version/pouches/:pouch_id/redirect", getPouchRedirect);
```
Comment (kshahzada) [inline src/services/packages/routes.ts:14]: I like this update

File: src/services/packages/logic.ts [hunk 3]
```diff
@@ -16,20 +16,73 @@ export const readPouch = async (pouch_id: string) => {
...
+
+export const writeRating = async (pouch_body: { form_response: { hidden: { package_id: string }, answers: object[] } }) => {
+    const { form_response } = pouch_body;
+    const airtable_template = {
+        record_id: form_response.hidden.package_id,
+        enjoyment_level: form_response.answers.find((answer: { field: { id: string }}) => (question_ids.enjoyment_level.includes(answer.field.id))),
+        rating: form_response.answers.find((answer: { field: { id: string }}) => (question_ids.rating.includes(answer.field.id))),
+        notes: form_response.answers.find((answer: { field: { id: string }}) => (question_ids.notes.includes(answer.field.id))),
+    }
+    const pouch_details = await write("Ratings", airtable_template);
```
Comment (kshahzada) [inline src/services/packages/logic.ts:44]: if you are going to use the "write" function and have a "writeRating" function, you should move all the ratings logic out of "write" to here

**Review by kshahzada — APPROVED** [review]
I'm approving but we should clean a couple of things up when we have a chance

**Comment by kshahzada** [comment]
Also, you need to turn on signed commits

File: src/services/packages/logic.ts (hunk 3 above)
Comment (mhwatson) [reply src/services/packages/logic.ts:44]: Yee i'll abstract this into a separate PR, fair call

**Pull request merged**